streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1171,4 +1156,9 @@ private void clearStandbyRecords() {
     private void refreshMetadataState() {
         streamsMetadataState.onChange(metadataProvider.getPartitionsByHostState(), metadataProvider.clusterMetadata());
     }
+
+    private static void createLogContext(final String logPrefix) {
```

Comment:
can probably in-line and remove this method if we pass `LogContext` into the `StreamThread` constructor

Reply:
Needed a `public static void` method to initialize static variable `log`. Passed `LogContext` into this. Kept the `log` variable as static because some of the static classes are using it: 
`RebalanceListener`, `AbstractTaskCreator`, `TaskCreator` etc.

Reply:
You should be able to pass the `LogContext` as a constructor param to `AbstractTaskCreator` etc where they are constructed in `create`.
Then `RebalanceListener` can use the same when it is constructed in `StreamThread`.
Then i think you can remove this method and the `log` variable won't need to be static?